src/PostgREST/Request/DbRequestBuilder.hs

```diff
@@ -152,21 +153,25 @@ findRel schema allRels origin target hint =
   case rel of
     []  -> Left $ NoRelBetween origin target
     [r] -> Right r
-    rs  ->
-      -- Return error if more than one relationship is found, unless we're in a
-      -- self reference case.
-      --
-      -- Here we handle a self reference relationship to not cause a breaking
-      -- change: In a self reference we get two relationships with the same
-      -- foreign key and relTable/relFtable but with different
-      -- cardinalities(m2o/o2m) We output the O2M rel, the M2O rel can be
-      -- obtained by using the origin column as an embed hint.
-      let [rel0, rel1] = take 2 rs in
-      if length rs == 2 && relLink rel0 == relLink rel1 && relTable rel0 == relTable rel1 && relFTable rel0 == relFTable rel1
-        then note (NoRelBetween origin target) (find (\r -> relType r == O2M) rs)
-        else Left $ AmbiguousRelBetween origin target rs
+    -- Here we handle a self reference relationship to not cause a breaking
+    -- change: In a self reference we get two relationships with the same
+    -- foreign key and relTable/relFtable but with different
+    -- cardinalities(m2o/o2m) We output the O2M rel, the M2O rel can be
+    -- obtained by using the origin column as an embed hint.
+    rs@[rel0, rel1]  -> case (relCard rel0, relCard rel1, relTable rel0 == relTable rel1 && relFTable rel0 == relFTable rel1) of
+      (O2M cons1, M2O cons2, True) -> if cons1 == cons2 then Right rel0 else Left $ AmbiguousRelBetween origin target rs
```

Comment:
So this is interesting - the pattern O2M / M2O seems not to be hit at all. Why?

And can we just delete the whole row? Or will this be changed as part of the issue you mentioned anyway?

Note, that this impossible branch was not there before - before it was only the `note` that was not touched. So I wonder whether we can improve on that.

Reply:
Or... are we just missing a test-case for a self-reference embedding "the other way around"?

Reply:
Ah, I now understood:

> It will be deleted when I fix that issue.

So the whole thing will not be there anymore at all. All good then :)

Reply:
> So this is interesting - the pattern O2M / M2O seems not to be hit at all. Why?

This could be because when building the DbStructure, the M2O rels are always built first. So when finding the rel with `filter` on the `Relation` list, the `M2O` will always be first and thus the O2M-M2O case will not happen.

> So the whole thing will not be there anymore at all.

Yeah, this will be removed anyway.